src/ui/actions/session.ts

```diff
@@ -229,67 +229,87 @@ export function createSocket(
         }
       }
 
-      const sessionId = await createSession(recordingId, loadPoint, experimentalSettings, {
-        onEvent: (event: ProtocolEvent) => {
-          if (features.logProtocolEvents) {
-            queueAction(eventReceived({ ...event, recordedAt: window.performance.now() }));
-          }
-        },
-        onRequest: (request: CommandRequest) => {
-          if (features.logProtocol) {
-            queueAction(requestSent({ ...request, recordedAt: window.performance.now() }));
-          }
-        },
-        onResponse: (response: CommandResponse) => {
-          if (features.logProtocol) {
-            const clonedResponse = { ...response, recordedAt: window.performance.now() };
-
-            if (isSourceContentsCommandResponse(clonedResponse)) {
-              // Must be a source contents entry. Shrink the source text just to minimize store size.
-              // It's rare that we would want to look at the source text in the Protocol Viewer,
-              // and at most we might want to see if the initial lines match or something.
-              clonedResponse.result = {
-                ...clonedResponse.result,
-                contents: clonedResponse.result.contents.slice(0, 1000),
-              };
+      const initialPausePoint = await getInitialPausePoint(ThreadFront.recordingId!);
+      const focusRegion =
+        initialPausePoint && "focusRegion" in initialPausePoint
+          ? initialPausePoint.focusRegion
+          : undefined;
```

Comment:
```suggestion
      const focusRegion = initialPausePoint?.focusRegion;
```

Reply:
I think the way I wrote it is necessary because `initialPausePoint()` can return one of two types:

`{point: string, time: number, focusRegion: TimestampedPointRange | null}`

or

`{point: string, time: number}`

Reply:
The code before was:
```js
const focusRegion =
  initialPausePoint && "focusRegion" in initialPausePoint
    ? initialPausePoint.focusRegion
    : undefined;
```

I think that's functionally equivalent to:
```js
const focusRegion = initialPausePoint?.focusRegion;
```

Isn't it?

Reply:
It isn't functionally equivalent because this type doesn't allow `focusRegion` to be optional:

`{point: string, time: number, focusRegion: TimestampedPointRange | null}`

The easy thing here is to just make `focusRegion` optional.

Reply:
Sorry @miriambudayr. Can you point out to me _what_ the functional difference is between those two statements? Because they should evaluate exactly the same. (Or are you saying TSC complains?)